src/Cli/dotnet/Telemetry/TelemetryFilter.cs

```diff
@@ -27,9 +27,17 @@ public IEnumerable<ApplicationInsightsEntryFormat> Filter(object objectToFilter)
 
             if (objectToFilter is ParseResult parseResult)
             {
-                var topLevelCommandName = parseResult[DotnetName]?.AppliedOptions?.FirstOrDefault()?.Name;
+                var topLevelCommandName = parseResult.RootCommandResult.Children?
+                    .FirstOrDefault(c => c.Token() != null && c.Token().Type.Equals(TokenType.Command))?
```

Comment:
`.FirstOrDefault(c => c.Token() != null && c.Token().Type.Equals(TokenType.Command))?` happened several times. It is better to extract it to an extension method or similar